mod_reforged/hooks/skills/special/night_effect.nut

```diff
@@ -0,0 +1,43 @@
+::Reforged.HooksMod.hook("scripts/skills/special/night_effect", function(q) {
+	q.m.HitChancePerTile <- -10;
+
+	q.getTooltip = @(__original) function()
+	{
+		local ret = __original();
+		foreach (entry in ret)
+		{
+			if (entry.id == 12 && entry.icon == "ui/icons/ranged_skill.png")
+			{
+				entry.text = "Has " + ::MSU.Text.colorizeValue(this.m.HitChancePerTile, {AddPercent = true}) + " chance to hit per tile of distance"
+			}
+		}
+		return ret;
+	}
+
+	// This should eventually be removed once we add a general line for all kinds of 'HitChancePerTile' for a ranged attack
```

Comment:
I'm not sure if I understand this comment. If it means that eventually we want to add a single combined line in hit factors for all HitChancePerTile for ranged attack, I disagree with that. Each effect should show its own effect in the hit factors. Therefore, if I have understood it correctly, then this comment should be removed.

Reply:
This reworked NIght Effect changes the shared vanilla `HitChanceAdditionalWithEachTile` variable in the character properties.
That variable is also touched by other effects like many shooting skills and the perk ballistics.

Because it is a shared variable it should be displayed __only once__ in the combat tooltip. 
The Vanilla representation of this variable is just the following term:
![image](https://github.com/Battle-Modders/mod-reforged/assets/2252464/31bfa6cc-f261-4a5c-bdbb-41a5ca84fa74)

Once we adjust that term to display the accurate penalty from distance there is no more need to display the penalty from the night debuff

Reply:
That's like saying anything that change `properties.MeleeSkill` e.g. Thrust, Unstoppable, Reach Advantage, and a ton of other effects should all be combined into a single `+x% hitchance` entry in hit factors. I completely disagree with this approach.

Anything that modifies `HitChanceAdditionalWithEachTile` ends up modifying the hitchance. Therefore, each such effect should show its own separate entry in the Hit Factors tooltip and show the exact amount that it is changing the hitchance by.

Hence, this comment is suggesting something that I believe we shouldn't do.

Reply:
I will remove the comment.